File: wss_demo/python/ws_public_demo.py
import json
import logging
import time
from datetime import datetime

import websocket

logging.basicConfig(filename='logfile_wrapper.log', level=logging.DEBUG,
                    format='%(asctime)s %(levelname)s %(message)s')

#now = datetime.now() #dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
#print("date and time =", dt_string)
prev_send_time = int(time.time() * 1000)
topic = "orderBook_200.100ms.BTCUSD"

trade_results = {}
size = {}
size_usdt = {}


def on_message(ws, message):
    data = json.loads(message)
    print(data)


def on_error(ws, error):
    print('we got error')
    print(error)
    print('print error complete')


def on_close(ws):
    print("### about to close please don't close ###")


def on_open(ws):
    print('opened')
    ws.send(json.dumps({"op": "subscribe", "args": [topic]}))


def on_pong(ws, *data):
    print('pong received')


def on_ping(ws, *data):
    now = datetime.now()
    dt_string = now.strftime("%d/%m/%Y %H:%M:%S")
    print("date and time =", dt_string)
    print('ping received')


def connWS():
    ws = websocket.WebSocketApp(
        "wss://stream.bybit.com/realtime",
        on_message=on_message,
        on_error=on_error,
        on_close=on_close,
        on_ping=on_ping,
        on_pong=on_pong,
        on_open=on_open
    )
    ws.run_forever(
		#http_proxy_host='127.0.0.1',
		#http_proxy_port=1087,
		ping_interval=30,
		ping_timeout=10
	)


if __name__ == "__main__":
    websocket.enableTrace(True)
    connWS()

File: wss_demo/js/ws_public_demo.js
var url = require('url');
var WebSocket = require('ws');
var crypto = require('crypto');
var endpoint = "wss://stream-testnet.bybit.com/realtime"
console.log('attempting to connect to WebSocket %j', endpoint);
var client = new WebSocket(endpoint);
client.on('open', function () {
	console.log('"open" event!');
	console.log('WebSocket Client Connected');
	setInterval(()=>{client.ping()}, 30000);
	client.ping();
	client.send(JSON.stringify({"op": "subscribe", "args": ["orderBookL2_25.BTCUSD"]}));
});

client.on('message', function (data) {
	console.log('"message" event! %j', data);
});
client.on('ping', function (data, flags) {
	console.log("ping received");
});
client.on('pong', function (data, flags) {
	console.log("pong received");
});
